### PR TITLE
feat(auth): add selectors

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/selectors/auth.selector.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { DaffAuthLoginSelectors, daffAuthLoginSelectors } from './login.selector';
+
+
+export interface DaffAuthSelectors<T extends DaffAuthToken> extends DaffAuthRegisterSelectors<T>, DaffAuthLoginSelectors<T> {
+  selectAuthState: MemoizedSelector<object, DaffAuthReducerState>;
+  selectAuthLoading: MemoizedSelector<object, boolean>;
+  selectAuthErrors: MemoizedSelector<object, string[]>;
+}
+
+const createRegisterSelectors = <T extends DaffAuthToken>() => {
```
Comment (lderrickable) [inline libs/auth/src/selectors/auth.selector.ts:19]: This looks like it is misnamed

File: libs/auth/src/selectors/register.selector.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { createSelector, MemoizedSelector } from '@ngrx/store';
+
+import { selectAuthFeatureState } from './auth-feature.selector';
+import { DaffAuthToken } from '../models/auth-token';
+import {
+  DaffAuthRegisterReducerState,
+  DaffAuthFeatureState
+} from '../reducers/public_api';
+
+export interface DaffAuthRegisterSelectors<T extends DaffAuthToken> {
```
Comment (lderrickable) [inline libs/auth/src/selectors/register.selector.ts:10]: The `T extends DaffAuthToken` looks like it isn't needed

File: libs/auth/src/selectors/register.selector.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  );
+
+  return {
+    selectAuthRegisterState,
+    selectAuthRegisterLoading,
+    selectAuthRegisterErrors,
+  }
+};
+
+export const daffAuthRegisterSelectors = (() => {
```
Comment (lderrickable) [inline libs/auth/src/selectors/register.selector.ts:39]: We've decided on the `getDaff...Selectors` naming convention

File: libs/auth/src/selectors/auth.selector.ts (hunk 1 above)
Comment (griest024) [reply libs/auth/src/selectors/auth.selector.ts]: fixed

File: libs/auth/src/selectors/register.selector.ts (hunk 2 above)
Comment (griest024) [reply libs/auth/src/selectors/register.selector.ts]: removed the unneeded generic

File: libs/auth/src/selectors/register.selector.ts (hunk 3 above)
Comment (griest024) [reply libs/auth/src/selectors/register.selector.ts]: changed

File: libs/auth/src/selectors/register.selector.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  const selectAuthRegisterLoading = createSelector(
+    selectAuthRegisterState,
+    state => state.loading
+  );
+
+  const selectAuthRegisterErrors = createSelector(
+    selectAuthRegisterState,
+    state => state.errors
+  );
```
Comment (lderrickable) [inline libs/auth/src/selectors/register.selector.ts:29]: I think that these selectors should go in a different file too. I think we should only have selectors in the main file when we don't have a choice, i.e. when they are selectors that depend on other selectors within auth.

File: libs/auth/src/selectors/auth.selector.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  selectAuthErrors: MemoizedSelector<object, string[]>;
+}
+
+const createAuthSelectors = <T extends DaffAuthToken>() => {
+  const selectAuthState = createSelector(
+    getDaffAuthFeatureStateSelector<T>(),
+    state => state.auth
+  )
+
+  const selectAuthLoading = createSelector(
```
Comment (lderrickable) [inline libs/auth/src/selectors/auth.selector.ts:24]: I think that these selectors should go in a different file too. I think we should only have selectors in the main file when we don't have a choice, i.e. when they are selectors that depend on other selectors within auth.

File: libs/auth/src/selectors/auth.selector.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  );
+
+  return {
+    selectAuthState,
+    selectAuthLoading,
+    selectAuthErrors,
+  }
+};
+
+export const getDaffAuthSelectors = (() => {
```
Comment (lderrickable) [inline libs/auth/src/selectors/auth.selector.ts:41]: This needs to export the feature selector too

Comment (griest024) [reply libs/auth/src/selectors/auth.selector.ts]: added the feature selector to the main selector function

File: libs/auth/src/selectors/auth.selector.ts (hunk 5 above)
Comment (griest024) [reply libs/auth/src/selectors/auth.selector.ts:22]: added a new file to export all the selectors and only exported that new file

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**